internal/server/network/ovn/ovn_nb_actions.go

```diff
@@ -1255,14 +1255,18 @@ func (o *NB) UpdateLogicalSwitchDHCPv4Options(ctx context.Context, switchName OV
 
 	if opts.Router != nil {
 		dhcpOption.Options["router"] = opts.Router.String()
+	} else {
```

Comment:
I added these else branches for all the options for consistency, but they're probably not strictly in scope for this PR, I can move them to a new one if desired.

The delete is only really required for the dns servers (and the routes, to which I already added it in the PR for those) because otherwise they would never get set to empty if so configured.

The other options are always set anyways, but in case that ever changes it's probably nice if they all behave in the same way.